### DISCUSSION TITLE
what's the advantage of Database Binary storage format compared to something like Delta lake or Iceberg

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#3972] [General]
it is not specific to DuckDB, But I thought, I get a good chance for a technical answer.

is there any Technical advantage of using a DB storage file compared to something like Iceberg ? I know for OLTP you need a storage as you can't really write a lot of transaction using parquet files, but for an analytical Workload, what would be the advantage.

**Comment by danz68** [comment]
Hi there,

First of all, I am not part of duckdb development team. However, I did suggest and I was one of the active supporters of a single file storage solution (and one single file library, too). 

Being active in the OLTP domain for more than 20 years, with a couple of custom engines and single file OLTP solutions written by myself (but nowhere near the complexity and features of duckdb), I can say that a huge majority of the OLTP projects do not exceed several millions of related records to be investigated at once. A distributed system, while hardly required, requires a infrastructure design, maintenance and security which should never be the concern of any OLTP developer. The transportable format, single file maintenance, simple copy file backup solutions, already available OS file security would not require IT assistance. The success of SQLite, Firebird embedded, Excel files as non centralized single file solutions speak for themselves. With duckdb able to connect to external sources, the extensibility is covered to a certain degree. That degree would be enough for most of the workloads and projects configurations I can think of. 

It doesn’t mean a centralized solution should never be designed. Or that it doesn’t have its own advantages (automatic/scheduled data loading, backup, balanced load, granular access/security, collaborative work and development). But IMO, most of us need compact tools, simple interaction with IT infrastructures and easy to use communication protocols. And single file storage is a natural choice when it comes down to simplicity for any of these: development, transport, backup, independent work, security.
